### PR TITLE
bugfix: Completing the missing statistics

### PR DESCRIPTION
**Pull request by 3062**
### Issues associated with this PR

Your PR should present related issues you want to solve.

### Solutions
You should show your solutions about the issues in your PR, including the overall solutions, details and the changes. At this time, Chinese is allowed to describe these.

### UT result
Unit Test is needed if the code is changed, your unit test should cover boundary cases, corner cases, and some exceptional cases. And you need to show the UT result.

### Benchmark
If your code involves the processing of every request, you should give the Benchmark Result.

### Code Style
+ Make sure `Goimports` has run
+ Show `Golint` result

### CODE REVIEW

**Comment by taoyuanyuan** [comment]
fix other stats, and such as UpstreamConnectionClose.

File: pkg/filter/network/streamproxy/streamproxy.go [hunk 1]
```diff
@@ -237,14 +239,49 @@ func (p *proxy) onUpstreamEvent(event api.ConnectionEvent) {
 	case api.ConnectFailed:
 		p.requestInfo.SetResponseFlag(api.UpstreamConnectionFailure)
 	}
+
+	p.onUpstreamEventStats(event, event.IsClose() || event == api.ConnectTimeout)
```
Comment (taoyuanyuan) [inline pkg/filter/network/streamproxy/streamproxy.go:243]: we don't need to call `UpstreamConnectionActive.Dec` when the event is `api.ConnectTimeout`.

File: pkg/stream/http2/connpool.go [hunk 2]
```diff
@@ -80,7 +80,7 @@ func (p *connPool) NewStream(ctx context.Context, responseDecoder types.StreamRe
 		p.mux.Lock()
 		defer p.mux.Unlock()
 		if p.activeClient != nil && atomic.LoadUint32(&p.activeClient.goaway) == 1 {
-			p.activeClient = nil
+			p.deleteActiveClient()
```
Comment (taoyuanyuan) [inline pkg/stream/http2/connpool.go:83]: it will deadlock

**Review by doujiang24 — DISMISSED** [review]

**Review by taoyuanyuan — APPROVED** [review]
LGTM

**Review by doujiang24 — APPROVED** [review]

**Pull request merged**